src/commands/server/update.ts

```diff
@@ -182,65 +173,173 @@ Consider removing '--che-operator-image' to update Eclipse Che operator to the s
+          merge(crPatch, resetImagesCrPatch)
+          ctx[ChectlContext.CR_PATCH] = crPatch
         }
       }
     }
+  }
 
-    try {
-      await updateTasks.run(ctx)
-      await postUpdateTasks.run(ctx)
+  /**
+   * Check whether chectl should proceed with update.
+   * Asks user for confirmation (unless assume yes is provided).
+   * Is applicable to operator installer only.
+   * Returns true if chectl can/should proceed with update, false otherwise.
+   */
+  private async checkAbilityToUpdateCheOperatorAndAskUser(flags: any): Promise<boolean> {
+    const ctx = ChectlContext.get()
+    cli.info(`Existing Eclipse Che operator: ${ctx.deployedCheOperatorImage}`)
+    cli.info(`New Eclipse Che operator     : ${ctx.newCheOperatorImage}`)
 
-      this.log(getCommandSuccessMessage())
-    } catch (err) {
-      this.error(getCommandErrorMessage(err))
+    if (ctx.deployedCheOperatorImageName === DEFAULT_CHE_OPERATOR_IMAGE_NAME && ctx.newCheOperatorImageName === DEFAULT_CHE_OPERATOR_IMAGE_NAME) {
+      // Official images
+
+      if (ctx.deployedCheOperatorImage === ctx.newCheOperatorImage) {
+        if (ctx.newCheOperatorImageTag === 'nightly') {
+          cli.info('Updating current Eclipse Che nightly version to a new one.')
+          return true
+        }
+
+        if (flags[CHE_OPERATOR_CR_PATCH_YAML_KEY]) {
+          // Despite the operator image is the same, CR patch might contain some changes.
+          cli.info('Patching existing Eclipse Che installation.')
+          return true
+        } else {
+          cli.info('Eclipse Che is already up to date.')
+          return false
+        }
+      }
+
+      if (this.isUpgrade(ctx.deployedCheOperatorImageTag, ctx.newCheOperatorImageTag)) {
+        // Upgrade
+
+        const currentChectlVersion = getProjectVersion()
+        if (!ctx.isNightly && (ctx.newCheOperatorImageTag === 'nightly' || semver.lt(currentChectlVersion, ctx.newCheOperatorImageTag))) {
+          // Upgrade is not allowed
+          if (ctx.newCheOperatorImageTag === 'nightly') {
+            cli.warn(`Stable ${getProjectName()} cannot update stable Eclipse Che to nightly version`)
+          } else {
+            cli.warn(`It is not possible to update Eclipse Che to a newer version using the current '${currentChectlVersion}' version of chectl. Please, update '${getProjectName()}' to a newer version using command '${getProjectName()} update' and then try again.`)
+          }
+          return false
+        }
+
+        // Upgrade allowed
+        if (ctx.newCheOperatorImageTag === 'nightly') {
+          cli.info(`You are going to update Eclipse Che ${ctx.deployedCheOperatorImageTag} to nightly version.`)
+        } else {
+          cli.info(`You are going to update Eclipse Che ${ctx.deployedCheOperatorImageTag} to ${ctx.newCheOperatorImageTag}`)
+        }
+      } else {
+        // Downgrade
+
+        if (semver.gt(MIN_CHE_OPERATOR_INSTALLER_VERSION, flags.version)) {
```

Comment:
Should we check here installer type. If user used helm or olm installer, notify that these installers can't do downgrade at all?

Reply:
And what about helm installer type?

Reply:
This whole method is used with `operator` installer only.